pkg/controller/job/job_controller.go

```diff
@@ -393,7 +393,9 @@ func (jm *JobController) processNextWorkItem() bool {
 	}
 
 	utilruntime.HandleError(fmt.Errorf("Error syncing job: %v", err))
-	jm.queue.AddRateLimited(key)
+	if !errors.IsConflict(err) {
+		jm.queue.AddRateLimited(key)
```

Comment:
If status update failed, we still need to requeue this job; otherwise, the job status may never be updated.

Reply:
As a workaround, we can probably retry on update conflict. But in the long run we should not use # of requeues to compare against backoff limit.

Reply:
Yeah, I totally agree that the re-queues are not good for that. They're causing us more harm than good :(

Reply:
I've created https://github.com/kubernetes/kubernetes/issues/64787 to track this.